modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/NumericInTypeCoercionTest.java

```diff
@@ -0,0 +1,3267 @@
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_5_3_DOUBLE,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_5_3),
+                        ofTypeWithoutCast(defaultDecimal),
+                        castTo(defaultDecimal)
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_5_3_DECIMAL_5_0,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_5_3),
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_5_0)
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_5_3_DECIMAL_5_3,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_5_3),
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_5_3)
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_5_3_DECIMAL_6_1,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_5_3),
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_6_1)
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_5_3_DECIMAL_8_3,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_5_3),
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_8_3)
+                ),
+
+                // DECIMAL (6,1)
+
+                Arguments.of(
+                        NumericPair.DECIMAL_6_1_REAL,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_6_1),
+                        ofTypeWithoutCast(defaultDecimal),
+                        castTo(defaultDecimal)
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_6_1_DOUBLE,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_6_1),
+                        ofTypeWithoutCast(defaultDecimal),
+                        castTo(defaultDecimal)
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_6_1_DECIMAL_6_1,
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_6_1),
+                        ofTypeWithoutCast(defaultDecimal),
+                        ofTypeWithoutCast(Types.DECIMAL_6_1)
+                ),
+
+                Arguments.of(
+                        NumericPair.DECIMAL_6_1_DECIMAL_6_1,
```

Comment:
copy-paste near

Reply:
Thanks. Added a method that checks for duplicate pairs.